python/tvm/octo/utils/target_info.py

```diff
@@ -0,0 +1,172 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, wrong-import-position
+"""Utility functions for finding information about current device."""
+import os
+import regex as re
+import sys
+import tvm
+import psutil
+import subprocess
+import shutil
+import tvm
+
+
+def get_llvm_target() -> tvm.target.Target:
+    """Extract fully featured llvm target for current device.
+
+    Returns
+    -------
+    target : tvm.target.Target
+        A TVM target that fully describes the current devices CPU.
+    """
+    # If we cant find llc, we wont be able to extract more information.
+    if shutil.which("llc") is None:
+        return "llvm"
+
+    # Get host information from llc
+    stream = subprocess.run(["llc", "--version"], stdout=subprocess.PIPE)
+    cpu_info = stream.stdout.decode("utf-8")
+
+    # Parse out cpu line
+    cpu = re.search("(?<=Host CPU: ).+", cpu_info).group(0)
+
+    # Next extract attribute string.
+    platform = sys.platform
+    # Linux
+    if platform == "linux" or platform == "linux2":
+        stream = subprocess.run("lscpu", stdout=subprocess.PIPE)
+        feature_info = stream.stdout.decode("utf-8")
+        features = re.search("(?<=Flags: ).+", feature_info).group(0)
+        features = features.lower().strip().split(" ")
+        march = re.search("(?<=Architecture: ).+", feature_info).group(0).strip()
+        cores = re.search("(?<=Core\(s\) per socket: ).+", feature_info).group(0).strip()
+        sockets = re.search("(?<=Socket\(s\): ).+", feature_info).group(0).strip()
+        total_cores = str(int(cores) * int(sockets))
+        # Special case for x86_64 mismatch between underscore and hyphen
+        if march == "x86_64":
+            march = "x86-64"
+    else:
+        raise ValueError("Platform %s is not supported." % platform)
+
+    # Now we'll extract the architecture of the target.
+    stream = subprocess.run(["llc", "--version"], stdout=subprocess.PIPE)
+    march_info = stream.stdout.decode("utf-8")
+    # Remove header.
+    march_options = re.search("(?<=Registered Targets:).*", march_info, re.DOTALL).group(0)
+    march_list = []
+    for march_line in march_options.split("\n"):
+        if march_line != "":
+            march_list.append(march_line.strip().split(" ")[0])
+
+    valid_march = False
+    if march in march_list:
+        valid_march = True
+
+    # Build the base target.
+    host_target = (
+        subprocess.run(["llvm-config", "--host-target"], stdout=subprocess.PIPE)
+        .stdout.decode("utf-8")
+        .strip("\n")
+    )
+    target = "llvm -mcpu=%s -mtriple=%s -num_cores=%s" % (cpu, host_target, total_cores)
+
+    # If possible, add more attribute information.
+    if valid_march:
+        # Get list of valid attributes for the target architecture.
+        sp = subprocess.run(["llc", "-march=%s" % march, "-mattr=help"], stderr=subprocess.PIPE)
```

Comment:
here and elsewhere, also I noticed when running on an AMD EPYC (I lost my notes so I don't have details anymore), but using `-march=zenwhatever` in the final target string was almost twice as slow (90s vs 50s for some model) compared to `-march=core-avx2`. Is that expected?



```suggestion
        sp = subprocess.run(["llc", f"-march={march}", "-mattr=help"], stderr=subprocess.PIPE)
```

Reply:
We might have to add some checks for zenwhatever to use avx2 vectorization. I think with explicit mattr features though it should still work.